docs/bdd-introduction.md

```diff
@@ -0,0 +1,17 @@
+
+## Behavior Driven Development(BDD):
+
+   Behavior driven development (or BDD) is an software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
+
+   BDD focuses on obtaining a clear understanding of desired software behavior through discussion with stakeholders. It extends [TDD](docs/tdd-introduction) by writing test cases in a natural language that non-programmers can read. Behavior-driven developers use their native language in combination with the ubiquitous language of domain driven design to describe the purpose and benefit of their code. This allow the developers to focus on why the code should be created, rather than the technical details, and minimizes translation between the technical language in which the code is written and the domain language spoken by the business, users, stakeholders, project management, etc.
+     
+   Coming to reality, Customer is a person who is an expert in dealing the things other than Technology. They need not to be aware of Edge case scenarios,beta testing and Alpha testing and so on. Even if he(customer) is familiar of the Technology stack,the only thing that is evolving in much faster pace is business, which change at constant pace.
+     
+   In this modern world, teams understand the business change and structure their approach according to the business needs.Rather than attempt to conceive, build, and deliver the perfect application in one fell swoop, they steadily deliver work in small and usable chunks for every week continously.  The customer is encouraged to think in terms of what functionality will be delivered at the maximum benefit to the business and prioritize the pipeline of work accordingly.
+     
+   This is where BDD comes in.  It bakes quality into the process by smoothing communication between stakeholders.  
+  
+   **"Behavior-driven development is about implementing an application by describing its behavior from the perspective of its stakeholders”** -- Dan North
+   
+  The main intention behind BDD framework is to avoid communication gaps between team memebers, And have better understanding and communicating more often with real world examples.
```

Comment:
## suggestion:

The main intention of a BDD framework is to avoid communication gaps between team members, foster better understanding of the customer and promote continuous communication with real world examples.